### PR TITLE
Core: Deployment jdl support

### LINKED ISSUE #8547: Improve generate full stack app with jdl
##### **Overview of the issue**
The new generation feature of the full stack is cool but there are some bug / non-user friendly points.

- [x] 1. I first generate my stack with the `jhipster import-jdl bug.jdl` command. 
Next, i modify the jdl file, adding a `countryCode String` attribute to the Countries entity, or update an application.
I executes the command `jhipster import-jdl bug.jdl`.
Selecting an option in case of conflict is ultra complicated as the applications are generated in parallel. I do not necessarily want to use the `--force` option each time.

- [x] 2.add an option to update only the entities.which would be an equivalent of the option `--json-only'`and `for f in `ls .jhipster`; do jhipster entity $ {f%. *}; done`

- [x] 3. If I run the `jhipster import-jdl bug.jdl --json-only --ignore-application` command with or without making a change since the last command, , the generator is blocked after `Entity JSON files created . Entity generation skipped.`. obliged to do a CTRL + C.

- [x] 4. idem after modifying an entity the command `for f in `ls .jhipster`; do jhipster entity $ {f%. *} --force;`, the generator is blocked after `Congratulations, JHipster execution is complete!`

##### **Motivation for or Use Case**
Improve experience

##### **Reproduce the error**
`jhipster import-jdl bug.jdl`
updte an entity
`jhipster import-jdl bug.jdl`
try to choose an option for the conflict

##### **JHipster Version(s)**
5.4.2

##### **JHipster configuration**
<details><summary>jdl</summary>
application {
	config {
    	baseName gateway
        applicationType gateway
    }

    entities *
}
application {
	config {
    	baseName micro
        applicationType microservice
    }
    entities *
}

entity Region {
	regionName String
}

entity Country {
	countryName String
}

// an ignored comment
/** not an ignored comment */
entity Location {
	streetAddress String,
	postalCode String,
	city String,
	stateProvince String
}

entity Department {
	departmentName String required
}

/**
 * Task entity.
 * @author The JHipster team.
 */
entity Task {
	title String,
	description String
}

/**
 * The Employee entity.
 */
entity Employee {
	/**
	* The firstname attribute.
	*/
	firstName String,
	lastName String,
	email String,
	phoneNumber String,
	hireDate Instant,
	salary Long,
	commissionPct Long
}

entity Job {
	jobTitle String,
	minSalary Long,
	maxSalary Long
}

entity JobHistory {
	startDate Instant,
	endDate Instant,
	language Language
}

enum Language {
    FRENCH, ENGLISH, SPANISH
}

relationship OneToOne {
	Country{region} to Region
}

relationship OneToOne {
	Location{country} to Country
}

relationship OneToOne {
	Department{location} to Location
}

relationship ManyToMany {
	Job{task(title)} to Task{job}
}

// defining multiple OneToMany relationships with comments
relationship OneToMany {
	Employee{job} to Job,
	/**
	* A relationship
	*/
	Department{employee} to
	/**
	* Another side of the same relationship
	*/
	Employee
}

relationship ManyToOne {
	Employee{manager} to Employee
}

// defining multiple oneToOne relationships
relationship OneToOne {
	JobHistory{job} to Job,
	JobHistory{department} to Department,
	JobHistory{employee} to Employee
}

// Set pagination options
paginate JobHistory, Employee with infinite-scroll
paginate Job with pagination

// Use Data Transfert Objects (DTO)
// dto * with mapstruct

// Set service options to all except few
service all with serviceImpl except Employee, Job

// Set an angular suffix
// angularSuffix * with mySuffix

</details>

##### **Browsers and Operating System**
ubuntu 18.04

-   [x] Checking this box is mandatory (this is just to show you read everything)

### PR DESCRIPTION
**Pull request by deepu105**
Fix #8547

Sample 
```scala
deployment {
  deploymentType kubernetes
  appsFolders [store, invoice, notification, product]
  dockerRepositoryName "deepu105"
  serviceDiscoveryType no
  istio autoInjection
  istioRoute true
  kubernetesServiceType Ingress
  kubernetesNamespace jhipster
  ingressDomain "jhipster.192.168.99.100.nip.io"
}
```
- [ ] add docs

-   Please make sure the below checklist is followed for Pull Requests.

-   [ ] [Travis tests](https://travis-ci.org/jhipster/generator-jhipster/pull_requests) are green
-   [ ] Tests are added where necessary
-   [ ] Documentation is added/updated where necessary
-   [ ] Coding Rules & Commit Guidelines as per our [CONTRIBUTING.md document](https://github.com/jhipster/generator-jhipster/blob/master/CONTRIBUTING.md) are followed

### CODE REVIEW

File: generators/kubernetes/index.js [hunk 1]
```diff
@@ -81,75 +50,30 @@ module.exports = class extends BaseGenerator {
...
-                this.defaultAppsFolders = this.config.get('appsFolders');
-                this.directoryPath = this.config.get('directoryPath');
-                this.clusteredDbApps = this.config.get('clusteredDbApps');
-                this.serviceDiscoveryType = this.config.get('serviceDiscoveryType');
-                this.adminPassword = this.config.get('adminPassword');
-                this.jwtSecretKey = this.config.get('jwtSecretKey');
-                this.dockerRepositoryName = this.config.get('dockerRepositoryName');
-                this.dockerPushCommand = this.config.get('dockerPushCommand');
+            loadKubernetesConfig() {
+                // this.defaultAppsFolders = this.config.get('appsFolders');
```
[the remark this answers is not in the document]
Comment (deepu105) [reply generators/kubernetes/index.js]: yup done

File: generators/openshift/index.js [hunk 2]
```diff
@@ -87,74 +56,28 @@ module.exports = class extends BaseGenerator {
...
-                this.directoryPath = this.config.get('directoryPath');
-                this.clusteredDbApps = this.config.get('clusteredDbApps');
-                this.serviceDiscoveryType = this.config.get('serviceDiscoveryType');
-                this.monitoring = this.config.get('monitoring');
-                this.adminPassword = this.config.get('adminPassword');
-                this.jwtSecretKey = this.config.get('jwtSecretKey');
-                this.dockerRepositoryName = this.config.get('dockerRepositoryName');
-                this.dockerPushCommand = this.config.get('dockerPushCommand');
+            loadOpenshiftConfig() {
+                // this.defaultAppsFolders = this.config.get('appsFolders');
```
[the remark this answers is not in the document]
Comment (deepu105) [reply generators/openshift/index.js]: done

**Comment by deepu105** [comment]
@pascalgrimaud its weird, `jq` seems to fail on the build
```
*** Sonar results:
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0jq - commandline JSON processor [version 1.5]
Usage: jq [options] <jq filter> [file...]
	jq is a tool for processing JSON inputs, applying the
	given filter to its JSON text inputs and producing the
	filter's results as JSON on standard output.
	The simplest filter is ., which is the identity filter,
	copying jq's input to its output unmodified (except for
	formatting).
	For more advanced filters see the jq(1) manpage ("man jq")
	and/or https://stedolan.github.io/jq
	Some of the options include:
	 -c		compact instead of pretty-printed output;
	 -n		use `null` as the single input value;
	 -e		set the exit status code based on the output;
	 -s		read (slurp) all inputs into an array; apply filter to it;
	 -r		output raw strings, not JSON texts;
	 -R		read raw strings, not JSON texts;
	 -C		colorize JSON;
	 -M		monochrome (don't colorize JSON);
	 -S		sort keys of objects on output;
	 --tab	use tabs for indentation;
	 --arg a v	set variable $a to value <v>;
	 --argjson a v	set variable $a to JSON value <v>;
	 --slurpfile a f	set variable $a to an array of JSON texts read from <f>;
	See the manpage for more options.
100   371  100   371    0     0   3865      0 --:--:-- --:--:-- --:--:--  3905
(23) Failed writing body
```

**Comment by pascalgrimaud** [comment]
it means the build didn't manage to get data from Sonar.
Did you restart the build ? Because I can't see the full log

File: generators/docker-base.js [hunk 3]
```diff
@@ -18,21 +18,18 @@
...
- */
+const { getBase64Secret } = require('./utils');
+
 module.exports = {
     checkDocker: dockerUtils.checkDocker,
     checkImages,
     generateJwtSecret,
     configureImageNames,
-    setAppsFolderPaths
+    setAppsFolderPaths,
```
Comment (murdos) [inline generators/docker-base.js]: What do you think of merging all of `docker-base.js` functions into the new `generator-docker-base.js`?

File: generators/generator-docker-base.js [hunk 4]
```diff
@@ -0,0 +1,132 @@
...
+ */
+const chalk = require('chalk');
+const prompts = require('./docker-prompts');
+const BaseGenerator = require('./generator-base');
+const { loadFromYoRc, checkDocker, checkImages, generateJwtSecret, setAppsFolderPaths } = require('./docker-base');
+const statistics = require('./statistics');
+
+const constants = require('./generator-constants');
+
+module.exports = class extends BaseGenerator {
```
Comment (murdos) [inline generators/generator-docker-base.js]: You've named this file `generator-docker-base.js`, and I'm OK with it.
But I recently used `generator-base-blueprint.js` for the base generator for blueprint. For the sake of consistency, I think I should rename the blueprint one to `generator-blueprint-base.js`, shouldn't I?

Comment (deepu105) [reply generators/generator-docker-base.js]: no worries i'll rename to `generator-base-docker.js`, I think thats more uniform

**Comment by deepu105** [comment]
@pascalgrimaud I might need your help with the failing jq thing

**Comment by pascalgrimaud** [comment]
I'm having a look

**Comment by MathieuAA** [comment]
@pascalgrimaud If jq needs a file, it may have interpreted the output from curl as filter. Perhaps the simple `.`  filter will work

**Comment by pascalgrimaud** [comment]
it works for all other PR and not this one. That's why it's strange

**Comment by pascalgrimaud** [comment]
@deepu105 : I did a PR against your branch.
See https://travis-ci.org/pascalgrimaud/generator-jhipster/builds/453812861

**Comment by pascalgrimaud** [comment]
Still failing. I really don't understand why.
I'll remove `| jq` to see what happens

**Comment by pascalgrimaud** [comment]
@deepu105 : so the build doesn't fail any more. The reason why it failed was the curl to Sonar is empty:

```
*** Sonar results:
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100   371  100   371    0     0   4297      0 --:--:-- --:--:-- --:--:--  4313
```

It's very very strange. I'll have a deeper look later, but this PR can be merged now

**Comment by deepu105** [comment]
@jdubois this will need a minor release v5.7 (not patch)

**Comment by deepu105** [comment]
Docs done here https://github.com/jhipster/jhipster.github.io/pull/663

**Pull request merged**